### PR TITLE
Refactor maintenance related dev tasks

### PR DESCRIPTION
**Pull request by eduardoj**
Move maintenance related development Rails tasks to its own task namespace: `dev:maintenance:...`.

Make it easier to add tasks related to maintenance, such as creating different types of requests related to maintenance for the development environtment.

Now we can call:
- `rake dev:test_data:create`

as we did before, and it will add a basic maintenance environment.

### CODE REVIEW

File: src/api/lib/tasks/dev/maintenance.rake [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        title: 'official maintenance space',
+        target_project: update_project,
+        maintainer: admin,
+        commit_user: admin
+      )
+    end
+
+    desc 'Create a request with maintenance incident actions'
+    task request_with_incident_actions: :development_environment do
+      require 'factory_bot'
```
Comment (krauselukas) [inline src/api/lib/tasks/dev/maintenance.rake:28]: I think you have to require this also for the maintenance `:project` task, its also using the factories

**Review by krauselukas — CHANGES_REQUESTED** [review]
I would say if you extract the maintenance part, it should work on its own (so no dependencies to run other rake dev task upfront)

File: src/api/lib/tasks/dev/maintenance.rake [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    desc 'Set a maintenance environment'
+    task all: :development_environment do
+      Rake::Task['dev:maintenance:project'].invoke
+      Rake::Task['dev:maintenance:request_with_incident_actions'].invoke
+    end
+
+    desc 'Create a maintenance project'
+    task project: :development_environment do
+      admin = User.get_default_admin
+      leap = Project.find_by(name: 'openSUSE:Leap:15.0')
```
[the remark this answers is not in the document]
Comment (eduardoj) [reply src/api/lib/tasks/dev/maintenance.rake:14]: @krauselukas, I created the task `dev:test_data:create_leap_project` and made the dev:maintenance:project task dependent on the `dev:test_data:create_leap_project` one. This should fix the dependency problem.

File: src/api/lib/tasks/dev/maintenance.rake (hunk 1 above)
Comment (eduardoj) [reply src/api/lib/tasks/dev/maintenance.rake:31]: True, done.

**Comment by eduardoj** [comment]
> I would say if you extract the maintenance part, it should work on its own (so no dependencies to run other rake dev task upfront)

I created another commit to make the maintenance part work on its own, and also updated de description of the pull request. I tried both rake tasks (`dev:test_data:create` and `dev:maintenance:all`) after the changes and worked. Please have a look again.

File: src/api/lib/tasks/dev.rake [hunk 3]
```diff
@@ -234,14 +225,42 @@ namespace :dev do
...
+      # The interconnect doesn't work unless we set the distributions
+      FetchRemoteDistributionsJob.perform_now
+    end
+
+    desc 'Create Tumbleweed Repository'
+    task create_tw_repository: :create_interconnect do
+      require 'factory_bot'
+      include FactoryBot::Syntax::Methods
+      interconnect = Project.find_by(name: 'openSUSE.org', remoteurl: 'https://api.opensuse.org/public')
+      create(:repository, name: 'snapshot', project: interconnect, remote_project_name: 'openSUSE:Factory')
```
Comment (hennevogel) [inline src/api/lib/tasks/dev.rake:252]: Setting a repository on an interconnect makes little sense. Are we using this somewhere?

File: src/api/lib/tasks/dev.rake [hunk 4]
```diff
@@ -234,14 +225,42 @@ namespace :dev do
...
 
       # Create news
       Rake::Task['dev:news:data'].invoke
     end
+
+    desc 'Create Interconnect'
+    task create_interconnect: :development_environment do
+      require 'factory_bot'
+      include FactoryBot::Syntax::Methods
+      create(:remote_project, name: 'openSUSE.org', remoteurl: 'https://api.opensuse.org/public')
```
Comment (hennevogel) [inline src/api/lib/tasks/dev.rake:242]: Hm, do you expect people to call this outside of the `test_data:create` task? If so you probably should make sure that it works if there is already a project by that name...

**Comment by hennevogel** [comment]
I don't get it. Why would I want *only* test data for maintenance?

**Review by hennevogel — CHANGES_REQUESTED** [review]

File: src/api/lib/tasks/dev/maintenance.rake [hunk 5]
```diff
@@ -0,0 +1,68 @@
+namespace :dev do
```
Comment (hennevogel) [inline src/api/lib/tasks/dev/maintenance.rake:1]: I think this should still be below `test_data` and not `dev`

**Review by hennevogel — APPROVED** [review]

**Pull request merged**